src/bp/core/services/nlu/entities-service.ts

```diff
@@ -0,0 +1,104 @@
+import * as sdk from 'botpress/sdk'
+
+import { GhostService } from '..'
+
+import * as CacheManager from './cache-manager'
+import { IntentService } from './intent-service'
+
+const ENTITIES_DIR = './entities'
+
+const DUCKLING_ENTITIES = [
+  'amountOfMoney',
+  'distance',
+  'duration',
+  'email',
+  'number',
+  'ordinal',
+  'phoneNumber',
+  'quantity',
+  'temperature',
+  'time',
+  'url',
+  'volume'
+]
+
+export class EntityService {
+  private intentService!: IntentService
+
+  constructor(private ghostService: GhostService) {}
+
+  public load(intentService: IntentService) {
+    this.intentService = intentService
+  }
+
+  private sanitizeFileName(name: string): string {
+    return name
+      .toLowerCase()
+      .replace(/\.json$/i, '')
+      .replace(/[\t\s]/gi, '-')
+  }
+
+  private entityExists(botId: string, entityName: string): Promise<boolean> {
+    return this.ghostService.forBot(botId).fileExists(ENTITIES_DIR, `${entityName}.json`)
+  }
+
+  public getSystemEntities(): sdk.NLU.EntityDefinition[] {
+    return [...DUCKLING_ENTITIES, 'any'].map(e => ({
+      name: e,
```

Comment:
just being picky: `.map(name => ({ name, type: 'system'})`. Also I think we can move this outside of the class since it doesn't call 'this'